extensions/rich_text_components/Image/directives/OppiaNoninteractiveImageDirective.ts

```diff
@@ -43,15 +49,18 @@ oppia.directive('oppiaNoninteractiveImage', [
       controllerAs: '$ctrl',
       controller: ['$attrs', function($attrs) {
         var ctrl = this;
-        ctrl.filepath = HtmlEscaperService.escapedJsonToObj(
-          $attrs.filepathWithValue);
         ctrl.imageUrl = '';
         ctrl.loadingIndicatorUrl = UrlInterpolationService.getStaticImageUrl(
           LOADING_INDICATOR_URL);
         ctrl.isLoadingIndicatorShown = false;
         ctrl.isTryAgainShown = false;
+        var imageId = HtmlEscaperService.escapedJsonToObj(
+          $attrs.idWithValue);
 
         if (ImagePreloaderService.inExplorationPlayer()) {
+          var stateName = PlayerPositionService.getCurrentStateName();
+          ctrl.filepath = (
+            ExplorationEngineService.getImageFilePath(stateName, imageId));
```

Comment:
Move `getImageFilePath` to the `ImagePreloaderService` (which will internally check the activeState).